src/core/utils/HttpResponse/decorators.ts

```diff
@@ -40,10 +41,11 @@ export function decorateResponse(
   // Cookie forwarding is only relevant in the browser.
   if (typeof document !== 'undefined') {
     // Write the mocked response cookies to the document.
-    // Note that Fetch API Headers will concatenate multiple "Set-Cookie"
-    // headers into a single comma-separated string, just as it does
-    // with any other multi-value headers.
-    const responseCookies = init.headers.get('Set-Cookie')?.split(',') || []
+    // Temporary measure until jsdom environment is improved
+    // and getSetCookie of Headers can be used directly.
+    const responseCookies = HeadersPolyfill.prototype.getSetCookie.call(
```

Comment:
We shouldn't tailor to technologies that violate standard support. We should rely on `.getSetCookie()` because it's a supported method. If the underlying environment doesn't support it, it's the underlying environment that should be fixing it (i.e. including any necessary polyfills). **We mustn't rely on `headers-polyfill` anymore**. It's used only for its utilities around transforming headers. 

The only concern is what @mattcosta7 raised about the TypeScript supporting this method. It doesn't exist on earlier versions of TS:

- [4.7](https://www.typescriptlang.org/play?ts=4.7.4#code/DYUwLgBAFhC8EDsQHcIAkQEMAmIBOAzgBQCUAUGVAHQDm4AyuAMID2LA1gJYilA)
- [4.8](https://www.typescriptlang.org/play?ts=4.8.4#code/DYUwLgBAFhC8EDsQHcIAkQEMAmIBOAzgBQCUAUGVAHQDm4AyuAMID2LA1gJYilA)
- [4.9](https://www.typescriptlang.org/play?ts=4.9.5#code/DYUwLgBAFhC8EDsQHcIAkQEMAmIBOAzgBQCUAUGVAHQDm4AyuAMID2LA1gJYilA)
- [5.0](https://www.typescriptlang.org/play?ts=5.0.4#code/DYUwLgBAFhC8EDsQHcIAkQEMAmIBOAzgBQCUAUGVAHQDm4AyuAMID2LA1gJYilA)
- [5.1](https://www.typescriptlang.org/play?ts=5.1.6#code/DYUwLgBAFhC8EDsQHcIAkQEMAmIBOAzgBQCUAUGVAHQDm4AyuAMID2LA1gJYilA)

And seems to only land in [5.2](https://www.typescriptlang.org/play?ts=5.2.2#code/DYUwLgBAFhC8EDsQHcIAkQEMAmIBOAzgBQCUAUGVAHQDm4AyuAMID2LA1gJYilA).

The `Headers.prototype.getSetCookie` can also be used only [since Node.js 19](https://developer.mozilla.org/en-US/docs/Web/API/Headers/getSetCookie#browser_compatibility) (which translates to v20 for stable releases) while MSW supports v18.

With this in mind, we cannot ship `.getSetCookie()` method as a part of MSW—it will break for a lot of people because it won't exist type-wise and runtime-wise. 

We need to implement the logic/fix ourselves until TS 5.2 and Node.js v20 become the minimal supported versions, which is years in the future.

Reply:
## TL;DR

What you're doing is fine. Just the reasoning should be mentioned more prominently above this line. 

Can you please include a comment above this `HeadersPolyfill` usage briefly describing why we rely on a polyfill here? Thanks!

Reply:
FWIW - typescript not supporting this in 4.7-5.1 shouldn't matter for consumers using those versions, because this call is encapsulated from their tooling (it's internal and generally node_modules aren't typechecked except at their `.d.ts` level which shouldn't contain any information about this call to getSetCookie

But with neither node18, jest not vitest able to support this correctly, we'd basically break everyone's tests the second we shipped it which would be unideal. 

I'm very much onboard that jsdom needs to support this - although maybe one day people will test browser code in a browser....one day

We could also inline logic from the headerpolyfill here instead of using that directly, as another option

Reply:
> FWIW - typescript not supporting this in 4.7-5.1 shouldn't matter for consumers using those versions, because this call is encapsulated from their tooling

This is only true if they have `skipLibCheck: true`, which is not the default and many people still have it disabled, which will cause compilation problems. Granted, easy to fix but creates an unnecessary stress for consumers. 

> I'm very much onboard that jsdom needs to support this - although maybe one day people will test browser code in a browser....one day

Fingers crossed!

> We could also inline logic from the headerpolyfill here instead of using that directly, as another option to get away from the polyfill - it's just implemented there already

See no reason to duplicate code. We depend on `headers-polyfill` anyway right now, its implementation is tested, we should use it.